### ISSUE TITLE
initializeFlagScheme() should have an arg for a default for handleFlags()

### ISSUE THREAD

**Issue by dankelley** [dankelley/oce#1531]
At the moment, the various specialiizations of the `handleFlags()` generic function finds out a default by calling `defaultFlags()` on the object.  But all that does is to look up the name (in `@metadata$flagScheme$name`) and then test against pre-defined values. (See line 710 et seq. within `R/AllClass.R`.) This is not a good scheme when it comes to creating new packages that inherit flag methodologies to oce, because it would require an update of `oce` for each such package.  The better scheme would be for `initializeFlagScheme()` to have an extra argument, stating a reasonable default.

I know that this and #1530 are both a bit technical, and that my wording is probably not sufficiently specific for anyone reading this. But the path forward is very clear to me, and I can code it in such a way that previously-created objects will still work fine, so that the changes will not break any existing code. For this reason, I am likely to code these two things in tomorrow or the day after, without waiting my normal few days for feedback. Having said that, if another developer wants to weigh in, I'd be happy to hear ... so, any thoughts, @richardsc or @catevans?

**Comment by richardsc** [comment]
Was this ever coded "tomorrow or the day after" as in the comment above? Because if so, I think it would be a good addition to a new CRAN submission. But if not, it's up to the reporter to decide if it should be implemented in the next week or two.

**Comment by dankelley** [comment]
I didn't get around to doing this. This kind of low-level work is quite risky, without a robust test suite. We don't have a real-world test suite because these are early days of using oce classes as base classes for other projects.  If we were planning a CRAN release in August, I'd start in on this. But I think we don't have enough time to do this sort of thing for a release in a week or so, because releases tend to be about a year apart and I sure don't want to break any users' code in a CRAN release.

So, my decision is to delay this to after the CRAN submission.

Thanks for noticing this, @richardsc !

**Comment by dankelley** [comment]
I've reopened this and pinned it. I will consider closing sometime within a week or so.

**Comment by dankelley** [comment]
This issue has been ignored for almost a year, so I decided to address it in commit 622ab81429974b21dd444af8d39a841f7f5cdb45 of "develop", the commit notes for which provide some detail.  The test suite also has some checks on whether this works and they are worth looking at, as a way to understand how things work.

Also, I added an arg `update` that permits changing an existing scheme, as an option (still, though, not by default).  This will help with e.g. the https://github.com/argoCanada/argoFloats project, in which I had to do some nasty low-level work to get around the rules.
